moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -981,6 +1004,40 @@ bool TimeOptimalTrajectoryGeneration::computeTimeStamps(
     RCLCPP_ERROR(LOGGER, "It looks like the planner did not set the group the plan was computed for");
     return false;
   }
+
+  // Validate scaling
+  double velocity_scaling_factor = 1.0;
+  if (max_velocity_scaling_factor > 0.0 && max_velocity_scaling_factor <= 1.0)
+  {
+    velocity_scaling_factor = max_velocity_scaling_factor;
+  }
+  else if (max_velocity_scaling_factor == 0.0)
+  {
+    RCLCPP_DEBUG(LOGGER, "A max_velocity_scaling_factor of 0.0 was specified, defaulting to %f instead.",
+                 velocity_scaling_factor);
+  }
+  else
+  {
+    RCLCPP_WARN(LOGGER, "Invalid max_velocity_scaling_factor %f specified, defaulting to %f instead.",
+                max_velocity_scaling_factor, velocity_scaling_factor);
+  }
+
+  double acceleration_scaling_factor = 1.0;
+  if (max_acceleration_scaling_factor > 0.0 && max_acceleration_scaling_factor <= 1.0)
+  {
+    acceleration_scaling_factor = max_acceleration_scaling_factor;
+  }
+  else if (max_acceleration_scaling_factor == 0.0)
+  {
+    RCLCPP_DEBUG(LOGGER, "A max_acceleration_scaling_factor of 0.0 was specified, defaulting to %f instead.",
+                 acceleration_scaling_factor);
+  }
+  else
+  {
+    RCLCPP_WARN(LOGGER, "Invalid max_acceleration_scaling_factor %f specified, defaulting to %f instead.",
```

Comment:
Based on my reasoning above (assuming it makes sense to other people as well), it would make sense to keep this separate conditions since not defining the scaling factor could be commonplace whereas specificly defining it with a nonzero invalid value is definitely invalid. Regarding whether ERROR is used and this returns false is probably a matter of opinon. I think putting a warning and defaulting to 1 helps get the message across while keeping functionality working but I hear either approach.

Reply:
the zero and invalid conditions were merged to both be invalid so marking this as resolved